bag_verifier/src/main/scala/uk/ac/wellcome/platform/archive/bagverifier/builder/BagVerifierWorkerBuilder.scala

```diff
@@ -0,0 +1,115 @@
+package uk.ac.wellcome.platform.archive.bagverifier.builder
+
+import akka.actor.ActorSystem
+import com.amazonaws.services.s3.AmazonS3
+import com.typesafe.config.Config
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import uk.ac.wellcome.messaging.sns.SNSConfig
+import uk.ac.wellcome.messaging.typesafe.AlpakkaSqsWorkerConfigBuilder
+import uk.ac.wellcome.messaging.worker.monitoring.metrics.MetricsMonitoringClient
+import uk.ac.wellcome.platform.archive.bagverifier.services.s3.{
+  S3ReplicatedBagVerifier,
+  S3StandaloneBagVerifier
+}
+import uk.ac.wellcome.platform.archive.bagverifier.services.{
+  BagVerifierWorker,
+  ReplicatedBagVerifyContext,
+  StandaloneBagVerifyContext
+}
+import uk.ac.wellcome.platform.archive.common.ingests.services.IngestUpdater
+import uk.ac.wellcome.platform.archive.common.operation.services.OutgoingPublisher
+import uk.ac.wellcome.platform.archive.common.{
+  BagRootPayload,
+  ReplicaResultPayload
+}
+import uk.ac.wellcome.storage.{S3ObjectLocation, S3ObjectLocationPrefix}
+import uk.ac.wellcome.typesafe.config.builders.EnrichConfig._
+import uk.ac.wellcome.json.JsonUtil._
+
+object BagVerifierWorkerBuilder {
+  def buildBagVerifierWorker(config: Config)(
+    ingestUpdater: IngestUpdater[SNSConfig],
+    outgoingPublisher: OutgoingPublisher[SNSConfig]
+  )(
+    implicit s3: AmazonS3,
+    mc: MetricsMonitoringClient,
+    as: ActorSystem,
+    sc: SqsAsyncClient
+  ) = {
+    val isReplicatedBagMode =
+      config.getBoolean("bag-verifier.mode.replicated_bag")
```

Comment:
Right now we run two verifiers:

* Verifying the unpacked, standalone bag in S3
* Verifying the replicated bag in S3

Soon we’ll add:

* Verifying the replicated bag in Azure

I guess at that point, we could add a third value to this flag (and make it getString()) to determine the verifier behaviour? For now, maybe tweak the name so it’s a boolean? `bag-verifier.mode.is-replica`.

Something like `unpacked-s3`, `replica-s3`, `replica-azure`?

I like the flexibility it gives us. 👍